### DISCUSSION TITLE
Version finale ?

### DISCUSSION THREAD

**Discussion by Pebloop** [TempoDev/Arcade-Toulouse#43] [General]
Bon j'ai push des trucs, il reste 1 semaine et demi et au bout d'un moment faut faire des choix.
Alors j'ai rush un truc, s'il y a des problèmes, dites le maintenant, plus rien ne sera changé, sauf besoin vital après dimanche !

ps, je sais que la doc n'est pas à jour, j'essayerais de m'en occuper le plus vite possible.

**Comment by nadmax** [comment]
Bah en soit il y a des choses où ça met des erreurs exemple l'inclusion de l'interface Event dans IScene (c'est du détail mais je préfère le dire)

**Reply by Pebloop** [reply]
Ouais c'est possible, comme je l"ai dit j'ai vraiment rush le truc d'un coup en m'inspirant de ce qu'on avait dit au fil des discussions (et 2-3 trucs du fork que nasso avait partagé), mais c'est clairement pas une version stable ^^'

Si vous voyez des erreurs n'hésutez pas pour faire des issues et pull request !

**Reply by nadmax** [reply]
Oui alors attends, c'est parti !

**Reply by Jad31** [reply]
Tu n'as pas un groupe qui peut t'aider ? Tu as l'air de gérer le repo toute seule @TempoDev

**Reply by nadmax** [reply]
J'avoue @Ydos2 @justinefricou vous êtes où ?

**Reply by Ydos2** [reply]
J'ai réglé ces détails normalement

**Comment by nasso** [comment]
Je suis toujours contre `IComponent` !

**Reply by nadmax** [reply]
Ça serait quoi l'alternative ?

**Reply by nasso** [reply]
Ce que j'ai fais sur mon fork, à savoir [des structures toutes simples](https://github.com/nasso/Arcade-Toulouse/blob/master/src/IScene.hpp#L34).

**Reply by Ydos2** [reply]
Je pense que le IComponent est une bonne idée pour structurer de façon plus organiser les systèmes dans un obj.
Après oui je suis d'accord ton fork donne plus de simplicité mais on s'éloigne aussi du concept générique voulu a la base.

**Reply by nasso** [reply]
...quels systèmes?

**Reply by Ydos2** [reply]
J'ai utiliser le mauvais mot avec système mais cela ne change rien a ma remarque.
C'est plus orgniser d'utiliser les components directement.

**Comment by Jad31** [comment]
Hello, je n'ai pas très bien compris à quoi sers les forEach des interfaces IScene et IEntity :/
Quelles sont les use cases et pourquoi on devrait s'en servir ?

**Reply by Ydos2** [reply]
Le forEach est là pour parcourir les différents components, donc pour tous tes composants il va exécuter la fonction qu'il y a dedans.

**Reply by nasso** [reply]
Par exemple pour afficher les sprites, tu parcour les entités avec un `forEach`, puis pour chacune tu cherche un composant sprite (avec `getComponents` ou un `forEach`).

**Comment by Jad31** [comment]
Est-ce que la version actuelle est la version finale du repo ou va-t-il encore y a voir des changements ? Et est-ce que quelqu'un a déjà réussi à tout implémenter de manière fonctionnelle ?
Est-ce que quelqu'un pourrait également update la doc ? Je n'ai pas compris comment certaines choses fonctionnent :/ @TempoDev @Ydos2 @justinefricou

**Reply by Ydos2** [reply]
Salut !
Pour l'architecture je pense que oui c'est une version quasi finale (il y aura peut-être deux trois petites fonctions en plus).
Oui le groupe de Nassim vient de finir l'implémentation.
Le grand @nasso a dit qu'il aller faire une petite pull request pour la doc.

**Comment by Pebloop** [comment]
Bon, j'ai corrigé les erreurs du clavier et de exit qui est const (???). Si vous pullez faites attention a votre overide dans votre class scene et au nouveau nom du clavier.
